tools/uart_hw_test/uart_test_windows.c

```diff
@@ -0,0 +1,313 @@
+
+	BOOL result = ReadFile(handle, buf, len, &bytesRead, NULL);
+
+	if (result) {
+		return bytesRead;
+	}
+
+	return -1;
+
+}
+
+/* UART Rx loss check test
+ * 1. The target prints log on the console continuously
+ * 2. The ubuntu app sends data on the micom uart
+ * 3. The target verify rx data for data loss
+ */
+static void test_uart_rx(HANDLE handle, int baud)
+{
+	volatile int i = 0;
+	volatile unsigned char ch;
+
+	ch = UART_TEST_START_VAL;
+	WriteFile(handle, (void *)&ch, 1, NULL, NULL);
+
+	/* Send the test bytes for cnt times */
+	for (int x = 0; x < UART_TEST_COUNT; x++) {
+		ch = UART_TEST_VAL;
+		WriteFile(handle, (void *)&ch, 1, NULL, NULL);
+		ch = 0;
+		poll_read(handle, (void *)&ch, 1, HANDSHAKE_INTERVAL);
+		if (ch == UART_TEST_VAL) {
+			i++;
+		}
+	}
+
+	ch = UART_TEST_STOP_VAL;
+	WriteFile(handle, (void *)&ch, 1, NULL, NULL);
+	ReadFile(handle, (void *)&ch, 1, NULL, NULL);
+
+
+	if (UART_TEST_COUNT - i == 0) {
+		printf("[BAUD %6d Rx] PASS\n", baud);
+		pass++;
+	} else {
+		printf("[BAUD %6d Rx] FAIL Bytes lost = %d / %d\n", baud, UART_TEST_COUNT - i, UART_TEST_COUNT);
+		fail++;
+	}
+
+	PurgeComm(handle, PURGE_RXCLEAR);
+}
+
+/* UART Tx byte gap time test
+ * 1. The target prints log on the console continuously
+ * 2. The target sends data on the micom data uart
+ * 3. The ubuntu app checks tx packet gap using time api
+ */
+static void test_uart_tx(HANDLE handle, int baud)
+{
+    volatile unsigned char ch;
+	double tm_diff, max_diff = 0;
+	DWORD bytesRead = 0;
+	DWORD bytesWritten = 0;
+	LARGE_INTEGER freq, start, end;
+	int cnt = 0;
+	int numfail = 0;
+
+	// Get the performance counter frequency
+	QueryPerformanceFrequency(&freq);
+
+
+	int ret = 0;
+	do {
+		ret = poll_read(handle, (void *)&ch, 1, 100) ;
+		if (ret <= 0) {
+			printf("flag 1 , %d\n", ret);
+			numfail = UART_TEST_COUNT;
+			goto result;
+		}
+	} while (ch != UART_TEST_START_VAL);
+
+	/* Read bytes till we receive end of test value */
+	do {
+		QueryPerformanceCounter(&start);
+		if (poll_read(handle, (void *)&ch, 1, UART_POLL_TIMEOUT_MS) <= 0) {
+			printf("flag 2");
+			numfail += UART_TEST_COUNT - cnt;
+			goto result;
+		}
+		if (ch == UART_TEST_VAL) {
+			QueryPerformanceCounter(&end);
+			cnt++;
+
+			tm_diff = ((end.QuadPart - start.QuadPart) * 1000.0) / freq.QuadPart  ;
+			if (tm_diff > UART_MAX_DELAY_MS) {
+				printf("flag 3");
+				numfail++;
+				if (tm_diff > max_diff) {
+					max_diff = tm_diff;
+				}
+			}
+		}
+	} while (ch != UART_TEST_STOP_VAL);
+
+result:
+	if (numfail) {
+		fail++;
+		printf("[BAUD %6d Tx] FAIL (Num out of spec = %d / %d Max delay = %fms)\n", baud, numfail, UART_TEST_COUNT, max_diff);
+	} else {
+		pass++;
+		printf("[BAUD %6d Tx] PASS\n", baud);
+	}
+
+	PurgeComm(handle, PURGE_RXCLEAR);
+}
+
+static void handshake(HANDLE handle, char c)
+{
+	volatile char ch;
+	int ret;
+	do {
+		ch = 0;
+        WriteFile(handle, (void *)&c, 1, NULL, NULL);
+        ret = poll_read(handle, (void *)&ch, 1, HANDSHAKE_INTERVAL);
```

Comment:
Please check indentation. There are several errors in indentation. Please check them all.